src/net_processing.cpp

```diff
@@ -4559,22 +4933,23 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
 
             if (pto->m_tx_relay != nullptr) {
                 LOCK(pto->m_tx_relay->cs_tx_inventory);
+                LOCK(peer->m_recon_state_mutex);
```

Comment:
After building with both gcc 10.2.1 and clang 9, functional tests are failing with `AssertionError: Unexpected stderr Assertion failed: detected inconsistent lock order for 'peer->m_recon_state_mutex' in net_processing.cpp:4936 (in thread 'msghand')`